drop-transfer/src/file/mod.rs

```diff
@@ -337,9 +340,15 @@ where
         reader.consume(n);
 
         total_n += n as u64;
-        if let Some(progress_cb) = progress_cb.as_mut() {
-            progress_cb(total_n).await;
+        from_last_event += n as u64;
+
+        if let (Some(progress_cb), Some(granularity)) = (progress_cb.as_mut(), event_granularity) {
+            if from_last_event >= granularity {
+                progress_cb(total_n).await;
+                from_last_event -= granularity;
```

Comment:
We expect granularity to be bigger. I saw that, also wrote in the comment on the config field.

Reply:
Should it also be enforced in the code?

Reply:
` // If set the checksum events will be emited for every checksum_events_granularity bytes
  // Default value is 256KB. Any value used here must be bigger than 256KB.`

Reply:
I am coming from the defensive side - if progress would change somehow and would emit larger chunks it could break this assumption. We can't really put an `assert` in here and just crash, so I think having a loop as a failsafe is the easiest option which also allows removing that part from the comment :)

Reply:
Agree

Reply:
I don't think this is a problem, the progress events can be missed sometimes and it does not need to be precise.
The thing is that the file is not guaranteed to be read in those 256kb chunks, it's just the buffer size given to the OS

Reply:
I am pretty sure Libdrop emits progress of 100%(bytes==total_bytes) always, and ther'es slight risk of not emitting 100% for checksum progress

Reply:
You can emit manually after the `validate()` call finishes. Plus we have this `ChecksumFinised` event.
Think of it from the UI perspective. You don't need all the information, the last one relevant is enough